app/views/search/common/_goods_nomenclature.html.erb

```diff
@@ -1,10 +1,16 @@
 <tr class="govuk-table__row">
   <td class="govuk-table__cell">
-    <a href="<%= commodity_path(goods_nomenclature) %>" aria-describedby="code-for-commodity-<%= goods_nomenclature.id %>">
-      <%= goods_nomenclature.description.html_safe %>
-      <span class="govuk-visually-hidden">
-        (code <%= goods_nomenclature.code %>)
-      </span>    
+    <% if goods_nomenclature.instance_of?(GoodsNomenclature) %>
+      <a href="<%= commodity_path(goods_nomenclature) %>" aria-describedby="code-for-commodity-<%= goods_nomenclature.id %>">
+    <% else %>
+      <a href="<%= polymorphic_path(goods_nomenclature) %>" aria-describedby="code-for-commodity-<%= goods_nomenclature.id %>">
+    <% end  %>
```

Comment:
To clean this we could:
`<a href="<%= goods_nomenclature_url(goods_nomenclature) %>" aria-describedby="code-for-commodity-<%= goods_nomenclature.id %>">`

and add a helper on `goods_nomenclature_path`

actually, there is already a `goods_nomenclature_path`, we could generalize that.

Reply:
I think we can actually get rid of our use of commodity path in a separate PR and then we can remove the conditional